src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -366,3 +367,30 @@ void CascadiaSettings::_RemoveHiddenProfiles()
         throw ::TerminalApp::SettingsException(::TerminalApp::SettingsLoadErrors::AllProfilesHidden);
     }
 }
+
+// Method Description:
+// - Ensures that every profile has a valid "color scheme" set. If any profile
+//   has a colorScheme set to a value which is _not_ the name of an actual color
+//   scheme, we'll set the color table of the profile to something reasonable.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+// - Appends a SettingsLoadWarnings::UnknownColorScheme to our list of warnings if
+//   we find any such duplicate.
+void CascadiaSettings::_ValidateAllSchemesExist()
+{
+    bool foundInvalidScheme = false;
+    for (auto& profile : _profiles)
+    {
+        if (!profile.ValidateColorScheme(_globals.GetColorSchemes()))

```

Comment:
does `GetColorSchemes` return a copy or a reference? This could be expensive.

Reply:
reference 👍